### PR TITLE
added support for Jetpack Markdown

### PR DESCRIPTION
**Pull request by drallgood**
For people using Jetpack's markdown solution

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
This is great, though I think we're going to want to wait until #42 is merged first. If you can help test that PR, that would be awesome.

**Comment by mAAdhaTTah** [comment]
We just merged that big PR. You're probably going to want to start this from scratch, as there were a ton of changes, but I would definitely merge this.

**Comment by drallgood** [comment]
Done :)

**Comment by mAAdhaTTah** [comment]
Great. I'll take a look at this and test later tonight.

File: lib/controller.php [hunk 1]
```diff
@@ -185,6 +185,14 @@ public function import_blob($blob) {
 			'_sha'         => $blob->sha,
 		) );
 
+		if(class_exists("WPCom_Markdown")) {
+			$wpcomMd = WPCom_Markdown::get_instance();
+			$args = $wpcomMd -> wp_insert_post_data($args, array(
+				"ID"           => $this->id,
```
Comment (mAAdhaTTah) [inline lib/controller.php]: The controller doesn't have an id property. Also, does the wpcomMd's `wp_insert_post_data` method actually perform the insert?

File: lib/controller.php [hunk 2]
```diff
@@ -185,6 +185,14 @@ public function import_blob($blob) {
...
+				"post_type"    => $args['post_type']
```
Comment (mAAdhaTTah) [inline lib/controller.php]: I don't think we're exporting the `post_type` so `$args['post_type']` will throw a Undefined Offset notice.

File: lib/controller.php (hunk 1 above)
Comment (drallgood) [inline lib/controller.php]: Thanks for catching this...
No. `wp_insert_post_data` just does the markdown conversion and puts everything in the right place (`post_content` vs. `post_content_filtered`

File: lib/controller.php (hunk 2 above)
Comment (drallgood) [inline lib/controller.php]: Ugh.. That is a problem. I need the type or the plugin won't be able to do things correctly
Can I use the `layout` attribute instead?

Comment (mAAdhaTTah) [inline lib/controller.php]: Yes, actually, my mistake; [layout](https://github.com/benbalter/wordpress-github-sync/blob/master/lib/post.php#L250) is the `post_type`, so use that.

**Comment by drallgood** [comment]
I realized, that the changes in controller really aren't necessary anymore as Jetpack is triggered after we update/insert the post

File: lib/post.php [hunk 3]
```diff
@@ -140,6 +140,11 @@ public function post_content() {
 
 		if ( function_exists( 'wpmarkdown_html_to_markdown' ) ) {
 			$content = wpmarkdown_html_to_markdown( $content );
+		} else if(class_exists("WPCom_Markdown")) {
```
Comment (benbalter) [inline lib/post.php]: Per the [WordPress style guide](https://make.wordpress.org/core/handbook/coding-standards/php/#space-usage), I believe we'll want to add some whitespace around the parenthesis before we merge.

**Comment by mAAdhaTTah** [comment]
@drallgood Not sure I follow, though I'm not that familiar with how the Jetpack Markdown module works. The import is going to bring in Markdown if you're using the module, so that needs to be converted, I think, unless it does the conversion when you save the Markdown content to `post_content_filtered`? Either way, I think something does need to happen on the way in, but do fill me in if I'm mistaken.

**Comment by drallgood** [comment]
Jetpack's markdown module hooks directly into the post creation/update:
https://github.com/Automattic/jetpack/blob/master/modules/markdown/easy-markdown.php#L117
I didn't know that either, but that's what I figured out. If you try to convert it before hand, it will just be converted back to HTML for some reason....

> The import is going to bring in Markdown if you're using the module, so that needs to be converted, I think, unless it does the conversion when you save the Markdown content to post_content_filtered? 
The plugin takes the markdown content from `post_content`and converts it to HTML. The HTML is then stored in `post_content` while the markdown gets saved as `post_content_filtered` for later use.

Someone else should probably confirm that.

File: lib/post.php (hunk 3 above)
Comment (drallgood) [inline lib/post.php]: Done.

Sorry, for the mess. I'm not developing much with PHP these days...

**Comment by drallgood** [comment]
Any progress?

**Comment by mAAdhaTTah** [comment]
I think I'll be able to get to it this weekend, sorry. I don't use the Jetpack module so I'd have to investigate it a bit more to make sure we're doing it right.

**Comment by mAAdhaTTah** [comment]
It looks like this works the way you said, with the Markdown module converting the md -> HTML for us. However, I think I found a bug with with regex that matches the Yaml frontmatter.

In any case, I'll figure that out and we can merge this, but do you think you'd be able to squash your commits? If not, I can merge it for you.

**Comment by drallgood** [comment]
I'm trying and failing ;)

**Comment by drallgood** [comment]
Nope... Guess it's easier if you just copy those 5 lines...

**Comment by mAAdhaTTah** [comment]
No worries, it's in. Thanks for your help!

**Pull request closed without merging**